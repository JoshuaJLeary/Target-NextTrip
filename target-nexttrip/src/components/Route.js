import React, { Component } from 'react';
import { connect } from 'react-redux';

class Route extends Component {
  // handles input value for nextTripInput.route
  handleRouteChange = (event) => {
    const { onChange, dispatch } = this.props;
    onChange({ route: event.target.value });
    dispatch({ type: 'GET_ROUTE_DIRECTION', payload: event.target.value });
  }

  render() {
    // loop through nextTripRoute and return each index as an option element
    let routeOptions = this.props.nextTripRoute.map(( route ) => {
      return (
        <option key={route.Route} value={route.Route}>
          {route.Description}
        </option>
      )
    })

    return (
      <div data-testid={'route'} className='inputSelectContainer'>
        <select className='inputSelect' onChange={this.handleRouteChange}>
          <option>Select Route</option>
          {routeOptions}
        </select>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  state
})

export default connect(mapStateToProps)(Route);