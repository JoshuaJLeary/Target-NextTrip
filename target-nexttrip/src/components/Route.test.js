import React from 'react';
import ReactDOM from 'react-dom';
import Route from './Route';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import { createStore, applyMiddleware } from 'redux';
import reducer from '../reducers/reduxStore';
import { Provider } from 'react-redux';
import createSagaMiddleware from 'redux-saga';

const mockRoute = [
  {
    Description: 'METRO Green Line',
    ProviderID: '8',
    Route: '902'
  }
];

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  reducer,
  (applyMiddleware(sagaMiddleware)),
);

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store}>
      <Route nextTripRoute={mockRoute}/>
    </Provider>, div);
});

it('renders Route correctly', () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <Route nextTripRoute={mockRoute}/>
    </Provider>);
  expect(getByTestId('route')).toHaveTextContent('METRO Green Line');
 });

it('matches snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Route nextTripRoute={mockRoute}/>
    </Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
