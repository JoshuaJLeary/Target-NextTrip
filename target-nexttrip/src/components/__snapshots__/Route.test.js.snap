// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="inputSelectContainer"
  data-testid="route"
>
  <select
    className="inputSelect"
    onChange={[Function]}
  >
    <option>
      Select Route
    </option>
    <option
      value="902"
    >
      METRO Green Line
    </option>
  </select>
</div>
`;
